Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -3279,11 +3280,30 @@ void WebPageProxy::processNextQueuedMouseEvent()
 
     LOG_WITH_STREAM(MouseHandling, stream << "UIProcess: sent mouse event " << eventType << " (queue size " << internals().mouseEventQueue.size() << ")");
     m_process->recordUserGestureAuthorizationToken(event.authorizationToken());
-    sendWithAsyncReply(Messages::WebPage::MouseEvent(m_mainFrame->frameID(), event, sandboxExtensions), [this, protectedThis = Ref { *this }] (WebEventType eventType, bool handled) {
+    CompletionHandler<void(WebEventType, bool, std::optional<RemoteMouseEventData>)> completionHandler = [this, protectedThis = Ref { *this }, sandboxExtensions = WTFMove(sandboxExtensions), &completionHandler] (WebEventType eventType, bool handled, std::optional<WebCore::RemoteMouseEventData> remoteMouseEventData) {
```

Comment:
Yeah, this isn't quite right.  When this is called, from inside the lambda completionHandler will point to bad stack memory.  But this is a way to get a recursive lambda if you know it's going to be called immediately.  I do something similar in iso2022JPEncode but it turns out I don't need 2 lines!

I think what we want instead is a function on RemotePageProxy and WebPageProxy that takes a std::optional<Vector<SandboxExtension::Handle>> and event information and no completion handler.  It calls sendWithAsyncReply, and in the completion handler it checks for a RemoteMouseEventData and, if present, it propagates the event and sandbox extensions to the WebPageProxy or RemotePageProxy.  Also, organize it to make sure we don't have duplicate code.